crates/ruff/src/rules/flake8_bugbear/rules/abstract_base_class.rs

```diff
@@ -123,12 +128,23 @@ pub fn abstract_base_class(
             && is_empty_body(body)
             && !is_overload(&checker.ctx, decorator_list)
         {
-            checker.diagnostics.push(Diagnostic::new(
+            let mut diagnostic = Diagnostic::new(
                 EmptyMethodWithoutAbstractDecorator {
                     name: format!("{name}.{method_name}"),
                 },
                 stmt.range(),
-            ));
+            );
+            if checker.patch(Rule::EmptyMethodWithoutAbstractDecorator) {
+                diagnostic.set_fix(Edit::insertion(
```

Comment:
Do we need to be concerned about _also_ importing the decorator if it's not in the file? Seems weird that fixing the offending code could actually break it.

Reply:
Yeah we do need to worry about this! If you take a look at `crates/ruff/src/rules/pyupgrade/rules/lru_cache_with_maxsize_none.rs`, we have this call to `get_or_import_symbol` which will either import the symbol if it's not already imported, or give you the correct reference (e.g., `import abc as foo; foo.abstractmethod`).

Reply:
@aacunningham Yep, the main reason this autofix wasn't implemented when B027 was added is because we did not have the `get_or_import_symbol` functionality implemented yet.

Reply:
Tbh that's a pretty good reason to pass on adding it 😃 

Let me take a look at that and squeeze it in here

Reply:
Aight, "borrowed" inspiration from `lru_cache_with_maxsize_none.rs` (wish I had found that one sooner). Looking good, plus it gave me a clean answer for handling that one `Option` value, converting it to a Result inside that `try_set_fix` closure.

Thanks y'all, that `get_or_import_symbol` really made that too easy.